datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -324,6 +331,29 @@ def handle_tls_warning(self):
         with disable_warnings_ctx(InsecureRequestWarning, disable=True):
             yield
 
+    @contextmanager
+    def context(self, ctx):
```

Comment:
If the use case is to having multiple RequestsWrapper with different configurations.

Do you think instead of using a context manager, we can create multiple instance (or clone) of RequestsWrapper?

Using context manager feel a bit hackish: we are modifying/patching the same instance of RequestsWrapper and restoring its state, feels like monkey patching.

Reply:
Done!